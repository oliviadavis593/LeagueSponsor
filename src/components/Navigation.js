import React from 'react';
import { NavLink } from 'react-router-dom';
import Logo from '../imgs/leaguelogo.png';

function Navigation() {
    return(
        <div className='ls-navigation'>
            <div className='ls-logo'>
                <NavLink to='/'>
                    <img src={Logo} alt='ls-logo' className='ls-logo' />
                </NavLink>
            </div>
            <nav className='nav'>
                <NavLink to='/leagueform' className='ls-league__form'>Add League</NavLink>
                <NavLink to='/sponsor' className='ls-sponsor'>Sponsor</NavLink>
            </nav>
        </div>
    )
}

export default Navigation; 