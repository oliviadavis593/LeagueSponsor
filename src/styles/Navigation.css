.nav {
    font-size: 25px; 
}

.ls-league__form {
    margin-right: 40px; 
}

.ls-league__form,
.ls-sponsor {
    text-decoration: none; 
    color: #233142;
}

.ls-league__form:hover,
.ls-sponsor:hover {
    color: #fbc531;
    transition: 0.7s; 
}

@media only screen and (min-width: 720px) {
    .ls-logo {
        width: 400px; 
        margin: 0 auto; 
    }

    .ls-league__form,
    .ls-sponsor {
        font-size: 32px; 
    }
}