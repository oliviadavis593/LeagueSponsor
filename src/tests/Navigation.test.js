import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navigation from '../components/Navigation';

describe('Navigation Component', () => {
    it(`renders without crashing`, () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><Navigation /></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', () => {
        const tree = renderer 
            .create(<BrowserRouter><Navigation /></BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})