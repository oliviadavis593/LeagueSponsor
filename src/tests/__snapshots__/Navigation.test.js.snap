// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component renders the UI as expected 1`] = `
<div
  className="ls-navigation"
>
  <div
    className="ls-logo"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        alt="ls-logo"
        className="ls-logo"
        src="leaguelogo.png"
      />
    </a>
  </div>
  <nav
    className="nav"
  >
    <a
      aria-current={null}
      className="ls-league__form"
      href="/leagueform"
      onClick={[Function]}
    >
      Add League
    </a>
    <a
      aria-current={null}
      className="ls-sponsor"
      href="/sponsor"
      onClick={[Function]}
    >
      Sponsor
    </a>
  </nav>
</div>
`;
